scanner_test.go

```diff
@@ -93,11 +92,27 @@ func dup(a []*pb.Result) []*pb.Result {
 	return b
 }
 
+func testCallClose(scan *hrpc.Scan, c *mock.MockRPCClient, scannerID uint64,
+	group *sync.WaitGroup) error {
+	s, err := hrpc.NewScanRange(scan.Context(), table,
+		nil, nil, hrpc.NumberOfRows(0), hrpc.CloseScanner(), hrpc.ScannerID(scannerID))
+	if err != nil {
+		return err
+	}
+	c.EXPECT().SendRPC(&scanMatcher{scan: s}).Do(func(arg0 interface{}) {
+		group.Done()
+	}).Return(&pb.ScanResponse{}, nil).Times(1)
+	return nil
+}
 func TestScanner(t *testing.T) {
 	ctrl := test.NewController(t)
 	defer ctrl.Finish()
 	c := mock.NewMockRPCClient(ctrl)
 
+	var wg sync.WaitGroup
```

Comment:
that's a bit of an odd way to use wait group. The test would stall if the SendRPC from Close isn't called and it would be better to fail the test instead. The fact that you've setup and expect call should already achieve that, so you don't need the wait group here.

Reply:
I may get a missing calls error if I don't set up the wait. I put it in to avoid flaky tests.

Reply:
uh I think this is do to the fact that we are calling the closing SendRPC in a goroutine. I never really liked that and it does make the tests less predictable and can potentially end up not closing the scanner in production (for example if program execution has finished). This can be resolved in another PR if you want to work on it.

Reply:
or can just drop `go` part in this review as it should be trivial (ignoring the error). The clients can themselves call Close() in a goroutine  if the really want it to be done asynchnously

Reply:
Yes. I understood why I get missing call errors and then put in the wait for test. If the test stalls, it is a test failure. We do expect it to be called.  BTWI don't think the go routine will cause leak if a program is gracefully shutdown. And it is OK to have some left for server to close in that exceptional case.

Reply:
And there are other tests using wait group too.